### PR TITLE
Rpm/foreman netbox dependency

### PR DESCRIPTION
**Pull request by dgoetz**
All dependencies for foreman_netbox, will create a second PR for foreman_netbox after merge.

### CODE REVIEW

**Comment by evgeni** [comment]
> Do I need to split this up further as rubygem-openssl is not finding rubygem-ipaddr which is part of the same PR?

I'm afraid so.

On the other hand, I wonder, isn't OpenSSL a built in gem?

**Comment by dgoetz** [comment]
I removed the dependency via patch and commented it in the spec, also dropped the openssl and ipaddr which was only a dependency for it.
Build looked good, thanks for the tip! 

Now repoclosure fails with:
```
Added downloaded_rpms repo from ./downloaded_rpms/rhel7
package: tfm-rubygem-netbox-client-ruby-0.5.6-1.el7.noarch from downloaded_rpms
unresolved deps:
  tfm-rubygem(faraday_middleware) < 1
  tfm-rubygem(faraday_middleware) >= 0.11
```
But this package is already there if I am correct: https://yum.theforeman.org/plugins/nightly/el7/x86_64/tfm-rubygem-faraday_middleware-0.13.1-3.el7.noarch.rpm
Any ideas?

**Comment by evgeni** [comment]
I do not, no.

**Comment by dgoetz** [comment]
Does package_manifests.yaml play a role here? Because there "rubygem-faraday_middleware" is in "foreman_proxy_plugin_packages" while all I packaged in this PR is in "plugin_scl_packages" and "repoclosure_lookaside_repos" does not point at the plugins repo, but "repoclosure_target_repos" does.

So if this is the cause, did I add it to the wrong section or needs faraday_middleware be moved now as it is also a dependency for another package now?

**Comment by evgeni** [comment]
probably, yeah. I'll have a look after lunch in detail.

**Comment by dgoetz** [comment]
Ok, with the explanation the message makes sense. Separate PR is already testing, will update this one after merge.

**Comment by evgeni** [comment]
Merged to other one, please rebase and enjoy ;)

**Comment by dgoetz** [comment]
It is green now! :smile:

**Comment by evgeni** [comment]
Nice, I'll have a look at the specs later, but if it builds… ;)

File: packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.github
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/Dockerfile
+%license %{gem_instdir}/LICENSE.txt
+%{gem_instdir}/VERSION
+%{gem_instdir}/bin
+%{gem_instdir}/dump.sql
```
Comment (evgeni) [inline packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec:96]:
```suggestion
%exclude %{gem_instdir}/dump.sql
```

File: packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.github
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/Dockerfile
```
Comment (evgeni) [inline packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec:92]:
```suggestion
%exclude %{gem_instdir}/Dockerfile
```

(or put it into doc below)

File: packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+%license %{gem_instdir}/LICENSE.txt
+%{gem_instdir}/VERSION
+%{gem_instdir}/bin
+%{gem_instdir}/dump.sql
+%{gem_libdir}
+%{gem_instdir}/netbox-client-ruby_rsa
+%{gem_instdir}/netbox-client-ruby_rsa.pub
+%{gem_instdir}/netbox.env
```
[the remark this answers is not in the document]
Comment (dgoetz) [reply packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec]: exluded

File: packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec (hunk 1 above)
Comment (dgoetz) [reply packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec]: excluded

File: packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec (hunk 2 above)
Comment (dgoetz) [reply packages/plugins/rubygem-netbox-client-ruby/rubygem-netbox-client-ruby.spec]: moved to doc as there was already other docker content

**Comment by dgoetz** [comment]
Thanks for the review, I already pushed the changes.

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
```console
% koji add-pkg --owner kojiadm foreman-plugins-nightly-rhel7 tfm-rubygem-dry-configurable tfm-rubygem-dry-container tfm-rubygem-dry-core tfm-rubygem-dry-equalizer tfm-rubygem-dry-inflector tfm-rubygem-dry-initializer tfm-rubygem-dry-logic tfm-rubygem-dry-types tfm-rubygem-dry-schema tfm-rubygem-dry-validation tfm-rubygem-faraday-detailed_logger tfm-rubygem-interactor tfm-rubygem-netbox-client-ruby
Adding 13 packages to tag foreman-plugins-nightly-rhel7
% koji add-pkg --owner kojiadm foreman-plugins-nightly-el8 rubygem-dry-configurable rubygem-dry-container rubygem-dry-core rubygem-dry-equalizer rubygem-dry-inflector rubygem-dry-initializer rubygem-dry-logic rubygem-dry-types rubygem-dry-schema rubygem-dry-validation rubygem-faraday-detailed_logger rubygem-interactor rubygem-netbox-client-ruby
Adding 13 packages to tag foreman-plugins-nightly-el8
```

**Comment by dgoetz** [comment]
So repoclosure failed now, but I hope this is because of the packages added with koji add-pkg already!

**Comment by evgeni** [comment]
No idea why it failed, but going to ignore it as the run before was green!

**Pull request merged**